### PR TITLE
Warn about/prevent multiple direct remote connections

### PR DESCRIPTION
**Pull request by mvollmer**
Demo: https://youtu.be/lvR2youiY4M

- [x] don't rely on logout resetting local storage. instead, let cockpit-ws report host addresses of valid cookies to login.html
- [x] config option
- [ ] design
- [ ] tests
- [ ] coverage

### CODE REVIEW

File: pkg/static/login.js [hunk 1]
```diff
@@ -338,6 +338,39 @@ import "./login.scss";
...
+
+        // Protect against outdated cur_machine values.
+        if (cur_machine == "localhost" && !window.location.pathname.startsWith("/="))
+            return;
+        if (cur_machine && cur_machine != "localhost" && window.location.pathname.startsWith("/=" + cur_machine))
+            return;
+
+        function redirect_to_current_machine() {
+            if (cur_machine == "localhost")
+                login_reload("/");
```
Comment (cockpituous) [inline pkg/static/login.js:356]: These 3 added lines are not executed by any test.

File: pkg/static/login.js [hunk 2]
```diff
@@ -338,6 +338,39 @@ import "./login.scss";
...
+            else
+                login_reload("/=" + cur_machine);
```
Comment (cockpituous) [inline pkg/static/login.js:358]: This added line is not executed by any test.

File: pkg/static/login.js [hunk 3]
```diff
@@ -338,6 +338,39 @@ import "./login.scss";
...
+        }
+
+        environment.page.allow_multi_host = true; // XXX
+
+        if (cur_machine) {
+            if (!environment.page.allow_multi_host)
+                redirect_to_current_machine();
```
Comment (cockpituous) [inline pkg/static/login.js:365]: This added line is not executed by any test.

File: pkg/static/login.js [hunk 4]
```diff
@@ -338,6 +338,33 @@ import "./login.scss";
         event.stopPropagation();
     }
 
+    function deal_with_multihost() {
+        // If we are currently logged in to some machine, but still
+        // end up on the login page, we are about to load resources
+        // from two machines into the same browser origin.
+
+        const logged_into = environment["logged-into"];
+        const cur_machine = logged_into.length > 0 ? logged_into[0] : null;
```
Comment (cockpituous) [inline pkg/static/login.js:347]: This added line is not executed by any test.

File: pkg/static/login.js [hunk 5]
```diff
@@ -338,6 +338,33 @@
...
+
+        function redirect_to_current_machine() {
+            if (cur_machine === ".")
+                login_reload("/");
```
Comment (cockpituous) [inline pkg/static/login.js:351]: These 3 added lines are not executed by any test.

File: pkg/static/login.js [hunk 6]
```diff
@@ -338,6 +338,33 @@
...
+            else
+                login_reload("/=" + cur_machine);
```
Comment (cockpituous) [inline pkg/static/login.js:353]: This added line is not executed by any test.

File: pkg/static/login.js [hunk 7]
```diff
@@ -338,6 +338,33 @@
...
+        }
+
+        if (cur_machine) {
+            if (!environment.page.allow_multi_host)
+                redirect_to_current_machine();
```
Comment (cockpituous) [inline pkg/static/login.js:358]: These 3 added lines are not executed by any test.

File: pkg/static/login.js [hunk 8]
```diff
@@ -338,6 +338,33 @@
...
+            else {
+                id("multihost-message").textContent = format(_("You are already connected to '$0' in this browser session. Connecting to other hosts will allow them to execute arbitrary code on each other. Please be careful."),
+                                                             cur_machine == "." ? "localhost" : cur_machine);
+                id("multihost-get-me-there").addEventListener("click", redirect_to_current_machine);
+                show('#multihost-warning');
```
Comment (cockpituous) [inline pkg/static/login.js:363]: These 4 added lines are not executed by any test.

File: pkg/static/login.js [hunk 9]
```diff
@@ -948,6 +977,8 @@
     }
 
     function login_reload (wanted) {
+        console.log("RELOAD", wanted);
```
Comment (cockpituous) [inline pkg/static/login.js:980]: This added line is not executed by any test.

**Comment by mvollmer** [comment]
Let's split this. Preventing: #20860, warning: #20861.

**Pull request closed without merging**